src/client-mixins/request-param.helper.ts

```diff
@@ -150,6 +154,37 @@ export class RequestParamHelpers {
     // Remove the query string
     return urlObject.href.slice(0, urlObject.href.length - urlObject.search.length);
   }
+
+  static applyRequestParametersToUrl(url: string, parameters: TRequestQuery) {
```

Comment:
[optional]
I'm not a fan of trying to parse the URL ourselves when we can leverage URL()

There are so many edge cases
 - ex. the ? could be in the # area (or in the auth area etc..) and therefore not be actually the start of the query string -
 
 I'm sure the code would be simpler like that
 
 ```
 const url= new URL('https://api.twitter.com/2/tweets/:id?query=string')
 url.pathname = url.pathname.replace(':id', 1234) // some more advanced magic there
return url.toString()

Reply:
Also, a quick description above would be nice :) (an example of what it does for instance), as the name is not crystal clear

Reply:
Yeah, it's a way more cleaned way :)
When I've built it, I thought new URL would crash before `:` is an invalid URL component, right? (but it seems to accomodate)